From ad51081de8fa91b16551cf8cbdad2c35ed091ec3 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Tue, 21 Feb 2012 18:05:14 +0800
Subject: [PATCH 1079/2463] ENGR00174916 MX6x, console output hang 20 seconds
 when system bootup

It is cause by fb driver to init HDMI PHY
when HDMI driver not register.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/video/mxc_hdmi.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/video/mxc_hdmi.c b/drivers/video/mxc_hdmi.c
index c415042..c13bc70 100644
--- a/drivers/video/mxc_hdmi.c
+++ b/drivers/video/mxc_hdmi.c
@@ -1701,7 +1701,8 @@ static void mxc_hdmi_cable_connected(struct mxc_hdmi *hdmi)
 static int mxc_hdmi_power_on(struct mxc_dispdrv_handle *disp)
 {
 	struct mxc_hdmi *hdmi = mxc_dispdrv_getdata(disp);
-	mxc_hdmi_phy_init(hdmi);
+	if (hdmi->fb_reg && hdmi->cable_plugin)
+		mxc_hdmi_phy_init(hdmi);
 	return 0;
 }
 
-- 
1.7.7.4

